### ISSUE TITLE
How to configure user authentication

### ISSUE THREAD

**Issue by diwenjing** [SnappyDataInc/snappydata#663]
I did not find gemfirexd.properties this configuration file in the installation directory

```
示例：使用BUILTIN身份验证的服务器和定位器
–- start a locator for peer discovery on port 3241 
-- with authentication; below specifies a builtin 
-- system user on the command-line itself
-- (snappydata.user.gem1=gem1) but it is recommended 
-- to be in gemfirexd.properties in encrypted form 
-- using the "snappy-shell encrypt-password" tool
snappy-shell rowstore locator start -peer-discovery-port=3241 -auth-provider=BUILTIN  -snappydata.user.gem1=gem1 -user=gem1 -password=gem1
–- start three servers using different client 
-- ports and using the above started locator
snappy-shell rowstore server start -dir=dir1 -locators=localhost:3241 -client-port=1528 -auth-provider=BUILTIN  -snappydata.user.gem1=gem1 -user=gem1 -password=gem1
snappy-shell rowstore server start -dir=dir2 -locators=localhost:3241 -client-port=1529 -auth-provider=BUILTIN  -snappydata.user.gem1=gem1 -user=gem1 -password=gem1
snappy-shell rowstore server start -dir=dir3 -locators=localhost:3241 -client-port=1530 -auth-provider=BUILTIN  -snappydata.user.gem1=gem1 -user=gem1 -password=gem1
–- check from the RowStore command shell
–- connect using the locator's client-port 
-- (default 1527) for load balanced
–- connection to one of the servers/accessors 
-- transparently and for reliable
-- failover in case the server/accessor goes down
snappy-shell
snappy> connect client 'localhost:1527;user=gem1;password=gem1';
–- add a new database user
snappy> call sys.create_user('snappydata.user.gem2', 'gem2');
–- check members
snappy> select ID, KIND from sys.members;
–- output will show four members with one as 
-- locator and three data stores
```
fllow this example to configuration my server,locator.lead, it will throw exception,what is default system user? How can I config it?

```
[root@cloudera1 sbin]# ./snappy-start-all.sh 
cloudera2: Starting SnappyData Locator using peer discovery on: cloudera2[9998], other locators: cloudera2:9998
cloudera2: Starting Thrift server for SnappyData at address cloudera2/192.168.15.123[1527]
cloudera2: Logs generated in /opt/snappydata/locator/snappylocator.log
cloudera2: SnappyData Locator pid: 20794 status: stopped
cloudera2: Error starting server process: 
cloudera2: AuthenticationFailedException: Authentication failed for cloudera2(20794)<v0>:49854: No such user 'GEM1' - See log file for details.
cloudera1: Starting SnappyData Server using locators for peer discovery: cloudera2:9998
cloudera1: Starting Thrift server for SnappyData at address cloudera1/192.168.15.122[1527]
cloudera1: Logs generated in /opt/snappydata/server/snappyserver.log
cloudera1: SnappyData Server pid: 34848 status: stopped
cloudera1: Error starting server process: 
cloudera1: GemFireConfigException: Unable to contact a Locator service (timeout=5000ms). Operation either timed out or Locator does not exist. Configured list of locators is "[cloudera2(null)<v0>:9998]". - See log file for details.
cloudera3: Starting SnappyData Leader using locators for peer discovery: cloudera2:9998
cloudera3: Logs generated in /opt/snappydata/lead/snappyleader.log
cloudera3: SnappyData Leader pid: 64382 status: stopped
cloudera3: Error starting server process: 
cloudera3: GemFireConfigException: Unable to contact a Locator service (timeout=5000ms). Operation either timed out or Locator does not exist. Configured list of locators is "[cloudera2(null)<v0>:9998]". - See log file for details.
```

**Comment by ymahajan** [comment]
I tried following steps and it worked fine for me. You can refer to these steps and see if it works for your case as well. There are some issues in the rowstore docs where the prefix used should be gemfirexd and not snappydata. You can create your own gemfirexd.properties under conf folder.

This was an old example with SnappyData 0.6 but should works for latest versions as well, what version of SnappyData are you using?  

snappydata-0.6-bin/conf $ cat gemfirexd.properties 
gemfirexd.auth-provider=BUILTIN 
mcast-port=0 
gemfirexd.user.locatoradmin=locatorpassword 
gemfirexd.user.serveradmin=serverpassword

snappydata-0.6-bin/bin $ ./snappy-shell locator start -dir=/home/ymahajan/snappydata-0.6-bin/locator -user=locatoradmin -password=locatorpassword 
Starting SnappyData Locator using peer discovery on: 0.0.0.0[10334] 
Starting DRDA server for SnappyData at address localhost/127.0.0.1[1527] 
Logs generated in /home/ymahajan/snappydata-0.6-bin/locator/snappylocator.log 
SnappyData Locator pid: 14201 status: running

/snappydata-0.6-bin/bin $ ./snappy-shell rowstore server start -dir=/home/ymahajan/snappydata-0.6-bin/server -user=serveradmin -password=serverpassword 
Starting SnappyData Server using multicast for peer discovery: 239.192.81.1[10334] 
Starting DRDA server for SnappyData at address localhost/127.0.0.1[1527] 
Logs generated in /home/ymahajan/snappydata-0.6-bin/server/snappyserver.log 
SnappyData Server pid: 14316 status: running

snappydata-0.6-bin/bin $ ./snappy-shell 
SnappyData version 0.6 
snappy> connect peer 'locators=localhost[10334];mcast-port=0;user=serveradmin;auth-provider=BUILTIN;gemfirexd.user.serveradmin=serverpassword;password=serverpassword;';

**Comment by diwenjing** [comment]
I use SnappyData0.9, Whether this example is available ?

**Comment by diwenjing** [comment]
My configueration files is as follows:
servers:
`cloudera1 -dir=/opt/snappydata/server -heap-size=4096m -locators=cloudera2:9998 -auth-provider=BUILTIN -user=admin -password=password`

locators:
`cloudera2 -peer-discovery-port=9998 -dir=/opt/snappydata/locator -heap-size=1024m -auth-provider=BUILTIN -user=admin -password=password`

leads:
`cloudera3 -dir=/opt/snappydata/lead -heap-size=4096m -J-XX:MaxPermSize=512m -spark.ui.port=9090 -locators=cloudera2:9998 -spark.executor.cores=10`

When I user the command sbin/snappy-start-all.sh with error:
```
cloudera2: Starting SnappyData Locator using peer discovery on: cloudera2[9998], other locators: cloudera2:9998
cloudera2: Starting Thrift server for SnappyData at address cloudera2/192.168.15.123[1527]
cloudera2: Logs generated in /opt/snappydata/locator/snappylocator.log
cloudera2: SnappyData Locator pid: 27564 status: stopped
cloudera2: Error starting server process: 
cloudera2: AuthenticationFailedException: Authentication failed for cloudera2(27564)<v0>:17040: No such user 'ADMIN' - See log file for details.
cloudera1: Starting SnappyData Server using locators for peer discovery: cloudera2:9998
cloudera1: Starting Thrift server for SnappyData at address cloudera1/192.168.15.122[1527]
cloudera1: Logs generated in /opt/snappydata/server/snappyserver.log
cloudera1: SnappyData Server pid: 3930 status: stopped
cloudera1: Error starting server process: 
cloudera1: GemFireConfigException: Unable to contact a Locator service (timeout=5000ms). Operation either timed out or Locator does not exist. Configured list of locators is "[cloudera2(null)<v0>:9998]". - See log file for details.
cloudera3: Starting SnappyData Leader using locators for peer discovery: cloudera2:9998
cloudera3: Logs generated in /opt/snappydata/lead/snappyleader.log
cloudera3: SnappyData Leader pid: 88107 status: stopped
cloudera3: Error starting server process: 
cloudera3: GemFireConfigException: Unable to contact a Locator service (timeout=5000ms). Operation either timed out or Locator does not exist. Configured list of locators is "[cloudera2(null)<v0>:9998]". - See log file for details.
```

**Comment by ashetkar** [comment]
@diwenjing Authentication and authorization for snappydata will be supported in 1.0
Currently, it works only in rowstore mode.

**Comment by diwenjing** [comment]
SnappyData 0.9 unsupported?

**Comment by ashetkar** [comment]
@diwenjing SnappyData 1.0 is not yet released. The latest released version is 0.9

**Comment by sumwale** [comment]
@diwenjing LDAP authentication and SQL compliant authorization via JDBC/ODBC/thrift APIs is available in the enterprise 1.0 RC release available from snappydata.io download page (final 1.0 release will be out later this week).
